discovery/ec2/ec2.go

```diff
@@ -245,6 +246,7 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 					ec2LabelInstanceID: model.LabelValue(*inst.InstanceId),
 				}
 				labels[ec2LabelPrivateIP] = model.LabelValue(*inst.PrivateIpAddress)
+				labels[ec2LabelPrivateDNS] = model.LabelValue(*inst.PrivateDnsName)
```

Comment:
couldn't `inst.PrivateDnsName` be `nil`?

Reply:
i'm not sure, i think if private ip address is set it should not be nil. i need to check this

Reply:
According to aws-sdk comments same conditions applied as for PublicDnsName.  But in case of PrivateDnsName private ip address is always assigned.

Reply:
We had a recent issue (https://github.com/prometheus/prometheus/issues/4684) where the EC2 service discovery crashed when used with EC2-like APIs so I'd prefer to be safe and add a check.

Reply:
Agree. I've added a check.